ffi/src/lib.rs

```diff
@@ -190,13 +204,53 @@ mod private {
     /// memory, but must only free it by calling [super::free_bool_slice]. Since the global
     /// allocator is threadsafe, it doesn't matter which engine thread invokes that method.
     unsafe impl Send for KernelBoolSlice {}
+    unsafe impl Send for KernelRowIndexArray {}
 
     /// # Safety
     ///
     /// If engine chooses to leverage concurrency, engine is responsible to prevent data races.
     unsafe impl Sync for KernelBoolSlice {}
+    unsafe impl Sync for KernelRowIndexArray {}
```

Comment:
You should add a `Safety` doc for this impl too

Reply:
Done